### DISCUSSION TITLE
404 when attempting to create transaction in 5.6.9

### DISCUSSION THREAD

**Discussion by ragnarkarlsson** [firefly-iii/firefly-iii#5471] [Firefly III] [answered]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)

### I've found a bug and checked that ...

- [X] ... [the documentation](https://docs.firefly-iii.org/) does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem
- [X] ... it's [definitely me, not you](https://github.com/firefly-iii/firefly-iii/blob/main/.github/its_you_not_me.md)

### Description

When attempting to create a new transaction (withdrawal/deposit) manually within Firefly-iii 5.6.9 (I'm running Docker install behind an nginx reverse proxy) this fails leading to a 404 error.

I was running 5.6.8 which exhibited this bug, updated to 5.6.9 and rebuilt the container but no change. Importing via the CSV importer (I've not migrated to the new data importer yet) worked fine.

### Expected behaviour

Result is a 404 with no transaction created.

Ticking the "After storing, return here to create another one." works as expected creating the transaction and returning to the page.

### Debug information

Debug information generated at 2022-01-02 14:54:32 Europe/London for Firefly III version **5.6.9**.

| Scope | Version |
| --- | --- |
| Firefly III | 5.6.9 |
| Firefly III API | 1.5.5 |
| Build | 66
, 02-01-2022 08:12:35 CET
 |
| PHP | 8.0.13 |
| Host | Linux |

| System info | Value |
| --- | --- |
| Installation ID | a69ae22b-e468-4fa6-bec4-a496669e26eb |
| Using docker? | true |
| Layout | v1 |
| System TZ | Europe/London |
| Browser TZ | Europe/London |
| App environment | local |
| App debug mode | false |
| App cache driver | file |
| App logging | notice, stack |
| Display errors | Off |
| Error reporting | ALL errors |
| Interface | apache2handler |
| Default language | en_GB |
| Default locale | equal |
| BCscale | 24 |
| DB drivers | mysql, pgsql, sqlite |
| Current driver | mysql |
| DB version | 18 (exp. 18) |
| Login provider | eloquent  |
| Trusted proxies (.env) | ** |

| User info | Value |
| --- | --- |
| Session start | 2022-01-01 00:00:00 |
| Session end | 2022-01-31 23:59:59 |
| Session first | 2010-04-05 00:00:00 |
| User ID | 1 |
| User language | en_US |
| User locale | en_US |
| Attempt at "en_US.utf8" | 'en_US.utf8' |
| Attempt at "en_US.UTF-8" | 'en_US.UTF-8' |
| User agent | Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/96.0.4664.110 Safari/537.36 |

### Steps to reproduce

Adding a new transaction manually by following steps:

1. Click menu top right
2. Create new withdrawal / deposit
3. Enter details for transaction
4. Do not tick "After storing, return here to create another one."
5. Hit submit

### Additional info

[Sun Jan 02 14:42:56.814830 2022] [mpm_prefork:notice] [pid 1] AH00163: Apache/2.4.38 (Debian) configured -- resuming normal operations
[Sun Jan 02 14:42:56.814991 2022] [core:notice] [pid 1] AH00094: Command line: 'apache2 -D FOREGROUND'
192.168.1.100 - - [02/Jan/2022:14:44:14 +0000] "GET /undefined?transaction_group_id=7361&message=created HTTP/1.1" 404 2671 "https://firefly.rkwd.co.uk/transactions/create/withdrawal" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/96.0.4664.110 Safari/537.36"

**Comment by JC5** [comment]
I can’t replicate this on the demo site. I’m not sure how that url could 404.

The “return here” option doesn’t reload anything, it clears the form. I’m not sure what the cause could be, sorry.

**Comment by ragnarkarlsson** [comment]
Hmm, similarly I'm unable to replicate it if I use my docker ip:port to access Firefly-iii direct, clearing my browser cache / local storage / cookies for the vanity URL doesn't help.

Which implies its an issue with my Nginx reverse proxy somehow? Not sure how, so I'm going to take a look at that now ...

**Comment by ragnarkarlsson** [comment]
Ok, so with the vanity url I end up at /undefined?transaction_group_id=7367&message=created

However with the docker IP:port it goes to /transactions/withdrawal?transaction_group_id=7368&message=created and works as expected.

I have checked and can't see anything obvious in the nginx config. This is ... perplexing.

Looking at the logs I see:

1. AUDIT entry to store new transaction over API.
2. POST to /api/v1/transactions with vanityurl/transactions/create/withdrawal
3. GET for /undefined?transaction_group_id=...

Oddly it is actually creating the transaction, I was looking at the wrong account for test transactions.

Looking in the browser console I'm getting this via the Nginx proxy.

ServiceWorker registration failed:  TypeError: Failed to register a ServiceWorker for scope ('https://vanityurl/') with script ('https://vanityurl/serviceworker.js?v=5.6.9'): A bad HTTP response code (404) was received when fetching the script.

Could this indicate something my nginx config is resulting in a block on?

**Comment by ragnarkarlsson** [comment]
I stripped my nginx config right down, removing various proxy_hide_header defaults specifically for the firefly-iii vanity URL and the service worker now loads correctly. 

Still getting 404 errors for the /undefined? url on both withdrawals and deposits, despite the fact they create the relevant transaction.

**Comment by ragnarkarlsson** [comment, marked as answer]
Sorry @JC5 this one really bugged me 😆 

I set up a completely new docker container, new database, installed from scratch. Doubted its the data but figured I would test to be sure.

When I went direct to http://ip:port and add a withdrawal this is what is recorded in the Network tab in my browser

![image](https://user-images.githubusercontent.com/3290527/147890292-df54b533-a05f-4209-8712-97ac0104c583.png)

However if I went to https://vanityurl and do the exact same thing this is the difference

![image](https://user-images.githubusercontent.com/3290527/147890265-aabe126f-36e3-4a8d-bc1c-77a81d649c1d.png)

I decided then to try on my Chromebook, and it worked perfectly ...

Following that I tried in Incognito mode also. Again, worked as intended.

This then led me to disable every Chrome extension and use the Empty cache & hard reload on the site. 

Tada, working. Switched back to my original docker container and tested again. Works.

I've now gone through enabling all of my extensions in Chrome and it works. No idea what caused this but its fixed for me.

**Reply by JC5** [reply]
Glad to hear it works!
